### PR TITLE
test(heading): migrate cypress tests to playwright

### PR DESCRIPTION
**Pull request by tomdavies73**
### Proposed behaviour

The Heading component should have in-browser tests using the Playwright framework

### Current behaviour

Heading currently has Cypress tests instead of Playwright tests

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

**Review by stephenogorman — APPROVED** [review]

**Comment by divyajindel** [comment]
Hi @tomdavies73 , Could you please remove mock component (HeadingComponent) from "heading-test.stories.tsx" file.

File: playwright/components/index.ts [hunk 1]
```diff
@@ -1,8 +1,12 @@
 import type { Page } from "@playwright/test";
-import { ICON } from "./locators";
+import { ICON, LINK } from "./locators";
 
 const icon = (page: Page) => {
   return page.locator(ICON);
 };
 
-export default icon;
+const link = (page: Page) => {
```
Comment (edleeks87) [inline playwright/components/index.ts:8]: question: Is it worth including a locator for `pill` and `help` here in this shared file as it's likely they'll be used in multiple test files?

File: src/components/heading/heading.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+  checkAccessibility,
+  getStyle,
+} from "../../../playwright/support/helper";
+
+const specialCharacters = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
+const testData = ["https://carbon.sage.com/"];
+const dividerAndSeparatorValue = [true, false];
+const headingType: HeadingType[] = ["h1", "h2", "h3", "h4", "h5"];
+
+test.describe("should render Heading component", () => {
```
Comment (edleeks87) [inline src/components/heading/heading.pw.tsx:31]: suggestion: I think these test descriptions could be a little more clearer so they act as better documentation etc

```suggestion
test.describe("Heading component", () => {
```

File: src/components/heading/heading.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+  specialCharacters.forEach((characterVals) => {
+    test(`should check children as ${characterVals} for Heading component`, async ({
```
Comment (edleeks87) [inline src/components/heading/heading.pw.tsx:33]:
```suggestion
    test(`should render with ${characterVals} passed as children`, async ({
```

File: src/components/heading/heading.pw.tsx [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+      page,
+    }) => {
+      await mount(<HeadingComponent> {characterVals} </HeadingComponent>);
+
+      await expect(headingPreview(page)).toContainText(characterVals);
+    });
+  });
+
+  specialCharacters.forEach((characterVals) => {
+    test(`should check title is ${characterVals} for Heading component`, async ({
```
Comment (edleeks87) [inline src/components/heading/heading.pw.tsx:44]:
```suggestion
    test(`should render with ${characterVals} passed as a title`, async ({
```

File: src/components/heading/heading.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+        page,
+      }) => {
+        await mount(<HeadingComponent subheader={characterVals} />);
+
+        await checkAccessibility(page);
+      });
+    });
+
+    specialCharacters.forEach((characterVals) => {
+      test(`should check accessibility when subtitleId is ${characterVals} for Heading component`, async ({
```
Comment (edleeks87) [inline src/components/heading/heading.pw.tsx:260]:
```suggestion
      test(`should pass accessibility checks when subtitleId is ${characterVals}`, async ({
```

**Review by edleeks87 — COMMENTED** [review]
Good work @tomdavies73, just some minor suggestions that I think would make the tests a little clearer to anyone viewing the file without any additional context

**Review by edleeks87 — DISMISSED** [review]

**Review by stephenogorman — DISMISSED** [review]

File: src/components/heading/heading.pw.tsx [hunk 6]
```diff
@@ -0,0 +1,363 @@
...
+      await mount(<HeadingComponent divider={dividerVals} />);
+
+      if (dividerVals) {
+        await expect(dividerPreview(page)).toHaveCSS(
+          "background",
+          "rgb(204, 214, 219) none repeat scroll 0% 0% / auto padding-box border-box"
+        );
+      } else {
+        expect(await dividerPreview(page).count()).toEqual(0);
+      }
```
Comment (Parsium) [inline src/components/heading/heading.pw.tsx:169]: **suggestion(non-blocking)**: I would suggest any tests that include conditional assertions should ideally be split into two separate tests for each condition:

```tsx
it("should render a divider when divider prop is passed", () => { ... });

it("should not render a divider when divider prop isn't passed", () => { ... });
```


so its a bit clearer if tests start failing in the console what functionality isn't working correctly.

**Comment by Parsium** [comment]
Nice one @tomdavies73! Just a few suggestions from me which are similar to @edleeks87's previous comments :)

**Review by stephenogorman — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by Parsium — APPROVED** [review]

**Pull request merged**